src/main/java/de/tum/in/www1/artemis/web/rest/UserResource.java

```diff
@@ -83,8 +87,29 @@ public ResponseEntity<List<UserDTO>> searchAllUsers(@RequestParam("loginOrName")
         if (loginOrName.length() < 3) {
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Query param 'loginOrName' must be three characters or longer.");
         }
+
+        User currentUser = userRepository.getUserWithGroupsAndAuthoritiesAndOrganizations();
+
+        Page<UserDTO> page;
         // limit search results to 25 users (larger result sizes would impact performance and are not useful for specific user searches)
-        final Page<UserDTO> page = userRepository.searchAllUsersByLoginOrName(PageRequest.of(0, 25), loginOrName);
+        if (currentUser.getOrganizations().size() > 0 && !currentUser.getAuthorities().contains(Authority.ADMIN_AUTHORITY)) {
+            page = new PageImpl<>(userRepository.searchAllUsersByLoginOrName(PageRequest.of(0, 25), loginOrName).stream().filter(user -> {
+                User userWithOrganizations = userRepository.findByIdWithGroupsAndAuthoritiesAndOrganizationsElseThrow(user.getId());
```

Comment:
This leads to a new database request for every user in the stream, so in the end there are up to 27 requests for one search.
Please instead add a new method to the UserRepository that filters directly in the database.

Reply:
Thank you for your fast response. I will change the code :)